File: 3.2/geom.erl
%%% -----------------------------------------------
%%% @author anders pearson
%%% @doc simple geometry module
%%% @end
%%% -----------------------------------------------
-module(geom).
-export([area/3]).

%% @spec area(Atom, X::integer(), Y::integer()) -> integer()
%% @doc calculates area of the given shape.
area(rectangle, X, Y) when X > 0, Y > 0 ->
		X * Y;
area(triangle, X, Y) when X > 0, Y > 0 ->
		(X * Y) / 2;
area(ellipse, X, Y) when X > 0, Y > 0 ->
		math:pi() * X * Y.

File: elixir/5.1/geom.ex
defmodule Geom do
	def area(shape, width, height) when width >= 0 and height >= 0 do
    case shape do
      :rectangle -> width * height
      :triangle -> width * height / 2.0
      :ellipse -> :math.pi() * width * height
      _ -> 0
    end
	end
end
